### PR TITLE
[include-what-you-use] some header fixups as recommended by iwyu

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Fixup some header inclusions as indicated by include-what-you-use

Followup PRs:
#19632 
#19631 
#19630 

## Motivation and context
Cleaning up includes can help with faster builds, less cruft, etc. Wanted to see how what include-what-you-use worked with the kodi codebase. 

Its a big report. Feel free to have a look at the one i generated against a linux build of kodi (~3MB) - ~~https://drive.google.com/file/d/1y4dmVC1LkrBEyb0nboqgQHDKVVAedbtF/view?usp=sharing~~
A little more refined report. Working towards mapping files for our libs/includes
https://drive.google.com/file/d/1AVPBbLc1JMr9vr1GxPkSXC1JODQfHADT/view?usp=sharing

the mappings used can be seen at https://github.com/fuzzard/xbmc/commits/iwyumapping

This is just the extreme tip of the iceberg from that report. Ive just picked some random files with fairly easy fixups.

I probably need to fix up some iwyu flags/mappings (spdlog/<bits/xx> etc) to be able to more easily use the automated apply tool of iwyu, so for now this is just done manually checking and applying suggested changes

Any recommendations of naming conventions for these sort of commits, let me know. I'll probably do some more as/when i get time/bored. Also feel free to pick your area of expertise and check/apply any fixups in the report above.

## How has this been tested?
buildtime

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [x] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/posix/filesystem/SMBDirectory.h [hunk 1]
```diff
@@ -8,10 +8,13 @@
 
 #pragma once
 
-#include "MediaSource.h"
-#include "SMBFile.h"
```
Comment (lrusak) [inline xbmc/platform/posix/filesystem/SMBDirectory.h:12]: probably need to move this to `SMBDirectory.cpp`

**Review by garbear — APPROVED** [review]
The changes look good to me. I think moving headers is important not for compile speed, but to clearly identify dependencies between files. Like how @lrusak noticed a dependency relationship that looks suspicious. I'd be open for more header fixups.

File: xbmc/platform/posix/filesystem/SMBDirectory.h (hunk 1 above)
Comment (fuzzard) [reply xbmc/platform/posix/filesystem/SMBDirectory.h]: Have just dropped this for now whilst i work through the ins and outs of iwyu

**Comment by fuzzard** [comment]
Have done a folder of changes for xbmc/events/* as a bit of a test.

I'll stop adding anything extra to this PR now. If people are happy for these sort of changes, or probably more likely, dont want these sort of changes let me know.

I'll try stay off peoples toes for their code areas, but i would suggest others have a look at the includes for stuff they deal with just as a bit of an insight. The report is in the original description for those interested. If anyone has any questions about it, hit me up on slack

**Review by garbear — APPROVED** [review]
New changes look good. I hope iwyu changes keep coming. I challenge you to format games/ and we'll see how my includes efforts compare to a tool.

**Comment by fuzzard** [comment]
Sure, I'll do that next. So far, @peak3d is on point with stuff he does, haha. I'll look into games next ;)

**Pull request closed without merging**